server/src/services/album.service.ts

```diff
@@ -164,33 +164,16 @@ export class AlbumService {
 
   async addAssets(auth: AuthDto, id: string, dto: BulkIdsDto): Promise<BulkIdResponseDto[]> {
     const album = await this.findOrFail(id, { withAssets: false });
-
     await this.access.requirePermission(auth, Permission.ALBUM_READ, id);
 
-    const existingAssetIds = await this.albumRepository.getAssetIds(id, dto.ids);
-    const notPresentAssetIds = dto.ids.filter((id) => !existingAssetIds.has(id));
-    const allowedAssetIds = await this.access.checkAccess(auth, Permission.ASSET_SHARE, notPresentAssetIds);
-
-    const results: BulkIdResponseDto[] = [];
-    for (const assetId of dto.ids) {
-      const hasAsset = existingAssetIds.has(assetId);
-      if (hasAsset) {
-        results.push({ id: assetId, success: false, error: BulkIdErrorReason.DUPLICATE });
-        continue;
-      }
-
-      const hasAccess = allowedAssetIds.has(assetId);
-      if (!hasAccess) {
-        results.push({ id: assetId, success: false, error: BulkIdErrorReason.NO_PERMISSION });
-        continue;
-      }
-
-      results.push({ id: assetId, success: true });
-    }
+    const results = await addAssets(
+      auth,
+      { accessRepository: this.accessRepository, repository: this.albumRepository },
+      { id, assetIds: dto.ids },
+    );
 
     const newAssetIds = results.filter(({ success }) => success).map(({ id }) => id);
     if (newAssetIds.length > 0) {
-      await this.albumRepository.addAssets({ albumId: id, assetIds: newAssetIds });
```

Comment:
It might be better to use find instead the way it would abort as soon as possible instead of going through the whole list.